VisualPinball.Unity/VisualPinball.Unity.Editor/Managers/ImageManager.cs

```diff
@@ -0,0 +1,232 @@
+		}
+
+		protected override List<ImageListData> CollectData()
+		{
+			List<ImageListData> data = new List<ImageListData>();
+
+			// collect list of in use textures
+			List<string> inUseTextures = new List<string>();
+			foreach (var item in _table.GetComponentsInChildren<IEditableItemBehavior>()) {
+				var texRefs = item.TextureRefs;
+				if (texRefs == null) { continue; }
+				foreach (var texRef in texRefs) {
+					var texName = GetMemberValue(texRef, item.ItemData);
+					if (!string.IsNullOrEmpty(texName)) {
+						inUseTextures.Add(texName);
+					}
+				}
+			}
+
+			foreach (var t in _table.Textures) {
+				var texData = t.Data;
+				data.Add(new ImageListData { TextureData = texData, InUse = inUseTextures.Contains(texData.Name)});
+			}
+
+			return data;
+		}
+
+		protected override void OnDataChanged(string undoName, ImageListData data)
+		{
+			OnDataChanged(undoName, data.TextureData);
+		}
+
+		protected override void AddNewData(string undoName, string newName) {
+			_table.Textures.SetNameMapDirty();
+			var sidecar = _table.GetComponentInChildren<TableSidecar>();
+			if (sidecar != null) {
+				Undo.RecordObject(sidecar, undoName);
+			}
+
+			var newTex = new Engine.VPT.Texture(newName);
+			_table.Textures.Add(newTex);
+			_table.Item.Data.NumTextures = _table.Textures.Count;
+		}
+
+		protected override void RemoveData(string undoName, ImageListData data)
+		{
+			_table.Textures.SetNameMapDirty();
+			var sidecar = _table.GetComponentInChildren<TableSidecar>();
+			if (sidecar != null) {
+				Undo.RecordObject(sidecar, undoName);
+			}
+
+			_table.Textures.Remove(data.Name);
+			_table.Item.Data.NumTextures = _table.Textures.Count;
+		}
+
+		private void OnDataChanged(string undoName, TextureData textureData)
+		{
+			RecordUndo(undoName, textureData);
+
+			// update any items using this tex
+			foreach (var item in _table.GetComponentsInChildren<IEditableItemBehavior>()) {
+				if (IsReferenced(item.TextureRefs, item.ItemData, textureData.Name)) {
+					item.MeshDirty = true;
+					Undo.RecordObject(item as UnityEngine.Object, undoName);
+				}
+			}
+		}
+
+		private void RecordUndo(string undoName, TextureData textureData)
+		{
+			if (_table == null) { return; }
+
+			// Run over table's texture scriptable object wrappers to find the one being edited and add to the undo stack
+			foreach (var tableTex in _table.Textures.SerializedObjects) {
+				if (tableTex.Data == textureData) {
+					Undo.RecordObject(tableTex, undoName);
+					break;
+				}
+			}
+		}
+
+		private void UpdateAllImages()
+		{
+			foreach (var t in _table.Textures) {
+				ReplaceImageFromPath(t.Data, t.Data.Path);
+			}
+		}
+
+		private void ReplaceImageFromPath(TextureData textureData, string path) {
+			if (_table == null || textureData == null || string.IsNullOrEmpty(path)) { return; }
+
+			byte[] newBytes = null;
+			try {
+				newBytes = File.ReadAllBytes(path);
+			} catch (Exception ex) {
+				Debug.LogException(ex);
```

Comment:
We have our own logger, it would be better to use that one.

```cs
private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
```

```cs
Logger.Error(ex);
```